daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Command.scala

```diff
@@ -15,30 +15,79 @@ sealed abstract class Command extends Product with Serializable {
 
 object Command {
 
+  /** Create a template, not by interface */
   final case class Create(
       templateId: Identifier,
       argument: SValue,
   ) extends Command
 
+  /** Create a template, by interface */
+  final case class CreateByInterface(
+      interfaceId: Identifier,
+      templateId: Identifier,
+      argument: SValue,
+  ) extends Command
+
+  /** Exercise a template choice, not by interface */
   final case class Exercise(
       templateId: Identifier,
       contractId: SContractId,
       choiceId: ChoiceName,
       argument: SValue,
   ) extends Command
 
+  /** Exercise a template choice, by interface */
+  final case class ExerciseByInterface(
+      interfaceId: Identifier,
+      templateId: Identifier,
+      contractId: SContractId,
+      choiceId: ChoiceName,
+      argument: SValue,
+  ) extends Command
+
+  /** Exercise an interface choice */
+  final case class ExerciseInterface(
+      interfaceId: Identifier,
+      contractId: SContractId,
+      choiceId: ChoiceName,
+      argument: SValue,
+  ) extends Command {
+    // TODO https://github.com/digital-asset/daml/issues/11342
+    //   The actual template id isn't known until run time.
+    //   The interface id is the best we've got.
+    val templateId = interfaceId
+  }
```

Comment:
```suggestion
  ) extends Command 
```
This seems unused. Better to remove it from `Command`

Reply:
It's used in the engine. https://github.com/digital-asset/daml/blob/main/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala#L120

Reply:
I missed it. I think it is OK for now.